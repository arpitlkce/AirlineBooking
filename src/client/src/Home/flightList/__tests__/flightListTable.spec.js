import React from 'react';
import renderer from 'react-test-renderer';
import FlightListTable from '../flightListTable';

describe('FlightListTable Component', () => {
    it('should render correctly in  mode', () => {
        let props = {
            'flightsData': [
                {
                    'flightData': {
                        'flightName': 'Go Air',
                        'flightNumber': 'G8-428'
                    },
                    'departData': {
                        'departureTime': '05:35',
                        'departureAirport': 'New Delhi'
                    },
                    'arrivalData': {
                        'arrivalTime': '08:35',
                        'arrivalAirport': 'Bangalore'
                    },

                    'duration': '2:30',
                    'price': '10,320',
                    'departureAirportCode': 'DEL',
                    'arrivalAirportCode': 'BLR'
                }]
        };

        const component = renderer.create(<FlightListTable items={props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
