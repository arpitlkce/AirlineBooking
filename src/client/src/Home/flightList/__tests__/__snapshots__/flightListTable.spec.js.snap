// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightListTable Component should render correctly in  mode 1`] = `
<div>
  <div
    className="ReactTable -striped -highlight transitTable"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "600px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
        >
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Flight
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Departure
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Arrival
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Duration
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Price
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "600px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -odd"
            role="row"
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="align-center"
              >
                <div>
                  Go Air
                </div>
                <div>
                  G8-428
                </div>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="align-center"
              >
                <div>
                  05:35
                </div>
                <div>
                  New Delhi
                </div>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="align-center"
              >
                <div>
                  08:35
                </div>
                <div>
                  Bangalore
                </div>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="align-center"
              >
                2:30
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="align-center"
              >
                10,320
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Book
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pagination-bottom"
    >
      <div
        className="-pagination"
      >
        <div
          className="-previous"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          className="-center"
        >
          <span
            className="-pageInfo"
          >
            Page
             
            <div
              className="-pageJump"
            >
              <input
                aria-label="jump to page"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="number"
                value={1}
              />
            </div>
             
            of
             
            <span
              className="-totalPages"
            >
              1
            </span>
          </span>
          <span
            className="select-wrap -pageSizeOptions"
          >
            <select
              aria-label="rows per page"
              onChange={[Function]}
              value={1}
            >
              <option
                value={5}
              >
                5 rows
              </option>
              <option
                value={10}
              >
                10 rows
              </option>
              <option
                value={20}
              >
                20 rows
              </option>
              <option
                value={25}
              >
                25 rows
              </option>
              <option
                value={50}
              >
                50 rows
              </option>
              <option
                value={100}
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          className="-next"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
